### PR TITLE
Fix/snowflake ext storage

### PR DESCRIPTION
**Pull request by mariq41**
### Brief summary of the change made
 This PR complements the need to parse variables in Azure external storage for URL, CREDENTIALS and  STORAGE_INTEGRATION.

### Are there any other side effects of this change that we should be aware of?
There are no side effects of the proposed change.

### Pull Request checklist
- [ ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -3862,7 +3862,7 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
             Ref("EqualsSegment"),
             Ref("QuotedLiteralSegment"),
             optional=True,
-        ),
+        ),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_snowflake.py:3865]:
```suggestion
        ),
```

File: test/fixtures/dialects/snowflake/create_stage.sql [hunk 2]
```diff
@@ -86,3 +86,11 @@ CREATE OR REPLACE STAGE your_stage_name
...
 CREATE OR REPLACE STAGE your_stage_name
   URL = 's3://your-s3-bucket/your-path-in-s3';
+
+CREATE STAGE mystage
+  URL=$your_variable
+  CREDENTIALS=(AZURE_SAS_TOKEN=$your_variable);
+
+CREATE STAGE mystage
+  URL=$your_variable
+  STORAGE_INTEGRATION=$your_variable;
```
Comment (WittierDinosaur) [inline test/fixtures/dialects/snowflake/create_stage.sql:96]: nit - please add a newline here

**Comment by IliyanKostov9** [comment]
Would kindly need your re-review again @WittierDinosaur :)

**Comment by WittierDinosaur** [comment]
Linting errors, please run black

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**